src/client.ts

```diff
@@ -600,6 +616,140 @@ export default class ConcordiumNodeClient {
         }
     }
 
+    async getRewardStatus(
+        blockHash: string
+    ): Promise<RewardStatus | undefined> {
+        if (!isValidHash(blockHash)) {
+            throw new Error('The input was not a valid hash: ' + blockHash);
+        }
+
+        type DateKey = KeysMatching<RewardStatusV1, Date>;
+        type BigIntKey = KeysMatching<RewardStatusV0 & RewardStatusV1, bigint>;
+
+        const dates: DateKey[] = ['nextPaydayTime'];
+        const bigInts: BigIntKey[] = [
+            'protocolVersion',
+            'gasAccount',
+            'totalAmount',
+            'totalStakedCapital',
+            'bakingRewardAccount',
+            'totalEncryptedAmount',
+            'finalizationRewardAccount',
+            'foundationTransactionRewards',
+        ];
+
+        const bh = new BlockHash();
+        bh.setBlockHash(blockHash);
+
+        const response = await this.sendRequest(
+            this.client.getRewardStatus,
+            bh
+        );
+
+        return unwrapJsonResponse<RewardStatus>(
+            response,
+            buildJsonResponseReviver(dates, bigInts),
+            intToStringTransformer(bigInts)
+        );
+    }
+
+    /**
+     * Retrieve list of bakers on the network.
+     * @param blockHash the block hash to get the smart contact instances at
+     * @returns A JSON list of baker IDs
+     */
+    async getBakerList(blockHash: string): Promise<BakerId[] | undefined> {
```

Comment:
When I try this, I get a parse error,
it tries to convert "0,1,2,3,4" to a Bigint :thinking: